arithmetization/src/main/java/net/consensys/linea/extradata/LineaExtraDataHandler.java

```diff
@@ -0,0 +1,143 @@
+    onStartup(blockchainService);
+  }
+
+  private void onStartup(final BlockchainService blockchainService) {
+    consumeExtraData(blockchainService.getChainHeadHeader());
+  }
+
+  @Override
+  public void onBlockAdded(final AddedBlockContext addedBlockContext) {
+    consumeExtraData(addedBlockContext.getBlockHeader());
+  }
+
+  private void consumeExtraData(final BlockHeader blockHeader) {
+    final var rawExtraData = blockHeader.getExtraData();
+
+    if (!Bytes.EMPTY.equals(rawExtraData)) {
+      for (final ExtraDataParser extraDataParser : extraDataParsers) {
+        if (extraDataParser.canParse(rawExtraData)) {
+          final var extraData = rawExtraData.slice(1);
+          extraDataParser.parse(extraData);
+          return;
+        }
+      }
+      log.warn("unsupported extra data field {}", rawExtraData.toHexString());
+    }
+  }
+
+  private interface ExtraDataParser {
+    boolean canParse(Bytes extraData);
+
+    void parse(Bytes extraData);
+
+    static Long toLong(final Bytes fieldBytes) {
+      return UInt32.fromBytes(fieldBytes).toLong();
+    }
+  }
+
+  @SuppressWarnings("rawtypes")
+  private class Version1Parser implements ExtraDataParser {
+    private final LineaProfitabilityConfiguration profitabilityConf;
+    private final FieldConsumer[] fieldsSequence;
+    private final MutableLong currFixedCostKWei = new MutableLong();
+    private final MutableLong currVariableCostKWei = new MutableLong();
+
+    public Version1Parser(final LineaProfitabilityConfiguration profitabilityConf) {
+      this.profitabilityConf = profitabilityConf;
+
+      final FieldConsumer fixedGasCostField =
+          new FieldConsumer<>(
+              "fixedGasCost", 4, ExtraDataParser::toLong, currFixedCostKWei::setValue);
+      final FieldConsumer variableGasCostField =
+          new FieldConsumer<>(
+              "variableGasCost", 4, ExtraDataParser::toLong, currVariableCostKWei::setValue);
+      final FieldConsumer minGasPriceField =
+          new FieldConsumer<>("minGasPrice", 4, ExtraDataParser::toLong, this::updateMinGasPrice);
+
+      this.fieldsSequence =
+          new FieldConsumer[] {fixedGasCostField, variableGasCostField, minGasPriceField};
+    }
+
+    public boolean canParse(final Bytes rawExtraData) {
+      return rawExtraData.get(0) == (byte) 1;
+    }
+
+    public synchronized void parse(final Bytes extraData) {
+      log.info("Parsing extra data version 1: {}", extraData.toHexString());
+      int startIndex = 0;
+      for (final FieldConsumer fieldConsumer : fieldsSequence) {
+        fieldConsumer.accept(extraData.slice(startIndex, fieldConsumer.length));
+        startIndex += fieldConsumer.length;
+      }
+
+      profitabilityConf.updateFixedAndVariableCost(
+          currFixedCostKWei.longValue(), currVariableCostKWei.longValue());
+    }
+
+    void updateMinGasPrice(final Long minGasPriceKWei) {
+      final var minGasPriceWei = Wei.of(minGasPriceKWei).multiply(1000);
```

Comment:
Can we move this constant somewhere where we can share its named value between profitability calculator and here?

Reply:
done